### ISSUE TITLE
[Feature] Improve Navigation of Large Amounts of Sub Directories

### ISSUE THREAD

**Issue by grravity** [SimpleMobileTools/Simple-Gallery#1987]
Been using Simple Gallery as my daily Gallery app for 2 years now, if there's one major issue I have, it's that I find myself constantly getting lose when navigating subdirectories.

I do not wish to have sub directories all listed in plain view, because it makes it harder to find the wanted directory when I need to scroll, so if I were to suggest something there it would be to be able to maybe adjust the size of the directory listing, or be able to view in some more condensed format without removing pictures.

However the main reason I come here, is to suggest a tree style view (for grouped directories), this would make it so much easier to see where you are when digging through subdirectories of subdirectories of directories, and navigate a lot quicker. I can't say exactly how I'd suggest to lay it out.

But overall, I really think there needs to be a navigation improvement to users with a bit more directories, and subdirectoriesc especially when they're grouped.

**Comment by tibbi** [comment]
Try enabling "Group direct subfolders" at the top Menu -> Change view type. Some improvements in that regard are on the roadmap, it cannot be customized for now.

**Comment by grravity** [comment]
Yes, that is what I was referring to in the second half when I kept mentioning "grouoed directories"

**Comment by tanooki-games** [comment]
Same problem here. I have some nested directories like:

```
Photos
+--------2020
              +-------2020-01
                           +-----------2020-01-01 New Yeat
                           +-----------2020-01-10 Birthday
              +-------2020-02
                           +-----------2020-02-29 20th Birthday of Grandpa
+---------2019
               + ------ ...

...
```

I could  not find a way to navigate in the gallery like I would in ON1, Lightroom, a normal file manager and so on. Please add this feature : )

Keep on the great work - Georg

**Comment by martin-holy** [comment]
Hi, can you please implement classical directory tree in a way that:
- only intrenal storage and SD card is visible in the root of folder browsing **OR** if there are set "included folders", you will see those as root folders
- navigating to folder will show all subfolders in that folder containing media (either just in that folder or any folder down the directory tree)
  for example if I navigate to folder "internal storage/folder A", I will see "folder A1" and "image 2" but not "folder A1a" and image 1
- also it will be helpful to show full path to current folder to see where I am and this path should be clickable to allow me to navigate to any folder in this path

Example without included folders:
- internal storage  <-- visible
  - folder A <-- visible
    - folder A1 <-- visible
      - folder A1a <-- visible
        - image 1 <-- visible
    - folder A2 <-- not visible
      - folder A2a <-- not visible
    - image 2 <-- visible
  - folder B <-- visible
    - folder B1 <-- visible
      - image 3 <-- visible
      
Example with included folders:
included folder are "internal storage/folder A/folder A1" and "internal storage/folder B"

- internal storage <-- not visible
  - folder A <-- not visible
    - folder A1 <-- visible
      - folder A1a <-- visible
        - image 1 <-- visible
    - folder A2 <-- not visible
      - folder A2a <-- not visible
    - image 2 <-- not visible
  - folder B <-- visible
    - folder B1 <-- visible
      - image 3 <-- visible
      
So this is what you will see with included folders.
Notice that "folder A1" moved one level up.
It is like every item in included folders creates new root, so you dont have to navigate long paths to get to your content.
- folder A1
  - folder A1a
    - image 1
- folder B
  - folder B1
    - image 3
    

I hope it is understandable. Thank you for your work on this useful application.

**Comment by StoopidoMan** [comment]
I will second this, I tried finding an alternative Gallery app that support sub-folders or folder hierarchy. It just makes organizing my albums much easier.

An example: I have a folder/album  called "Hunger" with many sub-folder/sub-albums with different photos of meals, like Pasta, pizza, etc. 
and because i have so many sub-folders, it is hard to find the photos i need. 
Hence the reason I uninstalled Simple Gallery, it is basically the same as Samsung Gallery app.

**Comment by StoopidoMan** [comment]
> 
> 
> Try enabling "Group direct subfolders" at the top Menu -> Change view type. Some improvements in that regard are on the roadmap, it cannot be customized for now.

I could't find "Group direct subfolders" would you share a screenshot. and which version are you using?

**Comment by tibbi** [comment]
"Group direct subfolders" is available only in the Pro app versions main folder screen, at the top Menu -> Change view type.

**Comment by giesmininkas** [comment]
Isn't this request similar to #11 ?

**Comment by darmbrust** [comment]
I think the default behavior, of just putting all photo folders at the top level if a fine way to start - but I would love the option to set a flag on a particular folder - which has a deep hierarchy - to treat just this folder as a tree.  So that none of the child folders from that particular folder would end up at the top level, we could just navigate it like a typical file browser would.  

In my use case, I sync in a significantly sized and highly organized photo tree from nextcloud onto my SD card.  Telling the gallery where that folder is leads to gallery becoming pretty useless, since I end up with dozens upon dozens of unorganized folders.  I've tried the subfolder grouping option... but I don't really understand what its trying to do.  The results don't make sense to me, anyway.    I've seen https://github.com/SimpleMobileTools/Simple-Gallery/issues/1886 - I don't have anything to add, other than the results of how it works now don't seem to make sense.
